src/compile/axis.ts

```diff
@@ -222,15 +226,14 @@ export function tickSizeEnd(model: Model, channel: Channel) {
   return undefined;
 }
 
-
 export function title(model: Model, channel: Channel) {
   const axis = model.axis(channel);
   if (axis.title !== undefined) {
     return axis.title;
   }
 
   // if not defined, automatically determine axis title from field def
-  const fieldTitle = fieldDefTitle(model.fieldDef(channel));
+  const fieldTitle = fieldDefTitle(axisFieldDef(model, channel));
```

Comment:
To be honest, I start wondering having `title` in fieldDef is really a good idea because of this.